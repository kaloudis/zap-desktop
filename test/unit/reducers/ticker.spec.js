// @flow

import Store from 'electron-store'
import tickerReducer, {
  SET_CURRENCY,
  SET_CRYPTO,
  GET_TICKERS,
  RECIEVE_TICKERS
} from 'reducers/ticker'

jest.mock('electron', () => {
  const { normalize } = require('path')

  return {
    app: {
      getPath: name => normalize(`/tmp/zap-test/${name}`),
      getAppPath: () => normalize('/tmp/zap-test')
    }
  }
})

Store.prototype.set = jest.fn()
Store.prototype.get = jest.fn()

describe('reducers', () => {
  describe('tickerReducer', () => {
    it('should handle initial state', () => {
      expect(tickerReducer(undefined, {})).toMatchSnapshot()
    })

    it('should have SET_CURRENCY', () => {
      expect(SET_CURRENCY).toEqual('SET_CURRENCY')
    })

    it('should have SET_CRYPTO', () => {
      expect(SET_CRYPTO).toEqual('SET_CRYPTO')
    })

    it('should have GET_TICKER', () => {
      expect(GET_TICKERS).toEqual('GET_TICKERS')
    })

    it('should have RECIEVE_TICKER', () => {
      expect(RECIEVE_TICKERS).toEqual('RECIEVE_TICKERS')
    })

    it('should correctly setCurrency', () => {
      expect(tickerReducer(undefined, { type: SET_CURRENCY, currency: 'foo' })).toMatchSnapshot()
    })

    it('should correctly setCrypto', () => {
      expect(tickerReducer(undefined, { type: SET_CRYPTO, crypto: 'foo' })).toMatchSnapshot()
    })

    it('should correctly getTicker', () => {
      expect(tickerReducer(undefined, { type: GET_TICKERS })).toMatchSnapshot()
    })

    it('should correctly receiveTicker', () => {
      expect(tickerReducer(undefined, { type: RECIEVE_TICKERS, ticker: 'foo' })).toMatchSnapshot()
    })
  })
})
