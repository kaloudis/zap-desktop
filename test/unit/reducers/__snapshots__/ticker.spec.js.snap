// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers tickerReducer should correctly getTicker 1`] = `
Object {
  "btcTicker": null,
  "crypto": "",
  "currency": "",
  "currencyFilters": Array [
    Object {
      "key": "btc",
      "name": "BTC",
    },
    Object {
      "key": "bits",
      "name": "bits",
    },
    Object {
      "key": "sats",
      "name": "satoshis",
    },
  ],
  "fiatTicker": "USD",
  "fiatTickers": Array [
    "USD",
    "EUR",
    "JPY",
    "GBP",
    "CAD",
    "KRW",
    "AUD",
    "BRL",
    "CHF",
    "CLP",
    "CNY",
    "DKK",
    "HKD",
    "INR",
    "ISK",
    "NZD",
    "PLN",
    "RUB",
    "SEK",
    "SGD",
    "THB",
    "TWB",
  ],
  "fromCurrency": "sats",
  "ltcTicker": null,
  "tickerLoading": true,
}
`;

exports[`reducers tickerReducer should correctly receiveTicker 1`] = `
Object {
  "btcTicker": undefined,
  "crypto": "",
  "currency": "",
  "currencyFilters": Array [
    Object {
      "key": "btc",
      "name": "BTC",
    },
    Object {
      "key": "bits",
      "name": "bits",
    },
    Object {
      "key": "sats",
      "name": "satoshis",
    },
  ],
  "fiatTicker": "USD",
  "fiatTickers": Array [
    "USD",
    "EUR",
    "JPY",
    "GBP",
    "CAD",
    "KRW",
    "AUD",
    "BRL",
    "CHF",
    "CLP",
    "CNY",
    "DKK",
    "HKD",
    "INR",
    "ISK",
    "NZD",
    "PLN",
    "RUB",
    "SEK",
    "SGD",
    "THB",
    "TWB",
  ],
  "fromCurrency": "sats",
  "ltcTicker": undefined,
  "tickerLoading": false,
}
`;

exports[`reducers tickerReducer should correctly setCrypto 1`] = `
Object {
  "btcTicker": null,
  "crypto": "foo",
  "currency": "",
  "currencyFilters": Array [
    Object {
      "key": "btc",
      "name": "BTC",
    },
    Object {
      "key": "bits",
      "name": "bits",
    },
    Object {
      "key": "sats",
      "name": "satoshis",
    },
  ],
  "fiatTicker": "USD",
  "fiatTickers": Array [
    "USD",
    "EUR",
    "JPY",
    "GBP",
    "CAD",
    "KRW",
    "AUD",
    "BRL",
    "CHF",
    "CLP",
    "CNY",
    "DKK",
    "HKD",
    "INR",
    "ISK",
    "NZD",
    "PLN",
    "RUB",
    "SEK",
    "SGD",
    "THB",
    "TWB",
  ],
  "fromCurrency": "sats",
  "ltcTicker": null,
  "tickerLoading": false,
}
`;

exports[`reducers tickerReducer should correctly setCurrency 1`] = `
Object {
  "btcTicker": null,
  "crypto": "",
  "currency": "foo",
  "currencyFilters": Array [
    Object {
      "key": "btc",
      "name": "BTC",
    },
    Object {
      "key": "bits",
      "name": "bits",
    },
    Object {
      "key": "sats",
      "name": "satoshis",
    },
  ],
  "fiatTicker": "USD",
  "fiatTickers": Array [
    "USD",
    "EUR",
    "JPY",
    "GBP",
    "CAD",
    "KRW",
    "AUD",
    "BRL",
    "CHF",
    "CLP",
    "CNY",
    "DKK",
    "HKD",
    "INR",
    "ISK",
    "NZD",
    "PLN",
    "RUB",
    "SEK",
    "SGD",
    "THB",
    "TWB",
  ],
  "fromCurrency": "",
  "ltcTicker": null,
  "tickerLoading": false,
}
`;

exports[`reducers tickerReducer should handle initial state 1`] = `
Object {
  "btcTicker": null,
  "crypto": "",
  "currency": "",
  "currencyFilters": Array [
    Object {
      "key": "btc",
      "name": "BTC",
    },
    Object {
      "key": "bits",
      "name": "bits",
    },
    Object {
      "key": "sats",
      "name": "satoshis",
    },
  ],
  "fiatTicker": "USD",
  "fiatTickers": Array [
    "USD",
    "EUR",
    "JPY",
    "GBP",
    "CAD",
    "KRW",
    "AUD",
    "BRL",
    "CHF",
    "CLP",
    "CNY",
    "DKK",
    "HKD",
    "INR",
    "ISK",
    "NZD",
    "PLN",
    "RUB",
    "SEK",
    "SGD",
    "THB",
    "TWB",
  ],
  "fromCurrency": "sats",
  "ltcTicker": null,
  "tickerLoading": false,
}
`;
